2_adventures/001/kotp/04_automated_testing_example/lib/fizz_buzz.rb

```diff
@@ -0,0 +1,19 @@
+module FizzBuzz
+  def self.play(number)
+    result = ''
+    result << 'Fizz' if number.divisible_by?(3)
+    result << 'Buzz' if number.divisible_by?(5)
+    result.empty? ? number : result
+  end
+end
+
+module FB_Tools
```

Comment:
I think I'd try to avoid more complicated default values for parameters like `rule = …` in the constructor or the range for `to_s`. I'd move `join` after the block in `what_is` to the last line, i.e. `result.join` because it's easy to miss after the `end`.